src/osd/ECCommon.cc

```diff
@@ -564,13 +559,29 @@ struct ClientReadCompleter : ECCommon::ReadCompleter {
         goto out;
       }
       bufferlist trimmed;
-      auto off = read.offset - aligned.first;
-      auto len = std::min(read.size, bl.length() - off);
-      dout(20) << __func__ << " bl.length()=" << bl.length()
-	       << " len=" << len << " read.size=" << read.size
-	       << " off=" << off << " read.offset=" << read.offset
-	       << dendl;
-      trimmed.substr_of(bl, off, len);
+      // If partial stripe reads are disabled read_offset_in_stripe will be 0
+      // which will mean trim_offset is 0. When partial reads are enabled the
+      // shards read (wanted_to_read) is a union of the requirements for each
+      // stripe, each range being read may need to trim unneeded shards
+      uint64_t read_offset_in_stripe = read.offset -
+	read_pipeline.sinfo.logical_to_prev_stripe_offset(read.offset);
+      uint64_t chunk_size = read_pipeline.sinfo.get_chunk_size();
+      uint64_t trim_offset = 0;
+      for (auto shard: wanted_to_read) {
```

Comment:
Range-based for loops are usually formatted with spaces on both sides of the colon:
```suggestion
      for (auto shard : wanted_to_read) {
```